service/worker/archiver/client.go

```diff
@@ -57,6 +60,7 @@ type (
 		logger        log.Logger
 		cadenceClient cclient.Client
 		numWorkflows  dynamicconfig.IntPropertyFn
+		rateLimiter   tokenbucket.TokenBucket
```

Comment:
You want this limit to be a dynamic config func() rather than a static integer. That way, you can pick a change to the ratelimit without having to restart the process. Take a look at this - which does something similar:
https://github.com/uber/cadence/blob/master/service/matching/ratelimiter.go

Reply:
Will do this in another PR.